File: app/src/main/java/com/fernandosierra/door2door/presentation/screens/routes/detail/RoutePresenter.java
package com.fernandosierra.door2door.presentation.screens.routes.detail;

import android.support.annotation.NonNull;

import com.fernandosierra.door2door.presentation.base.BasePresenter;

import javax.inject.Inject;

public class RoutePresenter extends BasePresenter<RouteView> {

    @Inject
    public RoutePresenter(@NonNull RouteView view) {
        super(view);
    }

    public void init() {
        secureViewCall(RouteView::init);
    }
}

File: app/src/main/java/com/fernandosierra/door2door/presentation/screens/routes/detail/RoutePresenter.kt
package com.fernandosierra.door2door.presentation.screens.routes.detail

import com.fernandosierra.door2door.presentation.base.BasePresenter
import javax.inject.Inject

class RoutePresenter @Inject constructor(view: RouteView) : BasePresenter<RouteView>(view) {

    fun init() = view { it.init() }
}